specs/flyout/flyout.md

```diff
@@ -0,0 +1,179 @@
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-popover`
+
+*Attributes*
+- `visible` - boolean, whether or not the Popover is visible, defaults to undefined, as this is controlled by the author
+- `trap-focus` - boolean, whether or not to keep focus contained inside the popover, defaults to true
+- `target` - string, the html id of the HTMLElement that the Popover is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `focus-target-on-close` - boolean, whether or not to focus back on to the _target element_ when the popover is closed, defaults to true
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic=
+- `delay` - number, the time delay before the popover is shown once triggered
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+    ** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Properties*
+- `targetElement` - the reference to the HTMLElement used as an target
+- `viewportElement` - the reference to the HTMLElement used as the viewport, required ot setup anchored region properly
+- `popoverVisible` - boolean, whether or not the Popover is visible
+
+*Slots*
+- default for content
+
+*Events*
+- `dismiss` - event fired when the Popover is dismissed
+
+*Functions*
+- `createPopoverTemplate(string: prefix)` - generates a `ViewTemplate` for the popover based on the given provided prefix string. This is required as popover uses an `anchored-region` internally and the create function generates a template using the appropriate `anchored-region` tag (ie "fast-anchored-region", "fluent-anchored-region"). Note that the appropriate `anchored-region` component must also be declared.
+
+*Position Options*
+
+- above-centered, below-centered
+
+![X](./images/ab-centered.png)
+
+- right-centered, left-centered
+
+![X](./images/rl-centered.png)
+
+- top-left-corner, top-right-corner, bottom-right-corner, bottom-left-corner
+
+![X](./images/corner-positions.png)
+
+- above-left-aligned, below-left-aligned
+
+![X](./images/ab-left-aligned.png)
+
+- above-right-aligned, below-right-aligned
+
+![X](./images/ab-right-aligned.png)
+
+- left-top-aligned, right-top-aligned
+
+![X](./images/rl-top-aligned.png)
+
+- left-bottom-aligned, right-bottom-aligned
+
+![X](./images/rl-bottom-aligned.png)
+
+### Anatomy and Appearance
+Parts:
+- `popover` - the popover itself, has `role=dialog`
+- `anchored-region` - the anchored region that controls the position of the popover
+
+
+*Template:*
+**ASK (2)**
+```html
+${when(
+    x => x.visible,
+    html<Popover>`
+        <div>
+            <fast-anchored-region
```

Comment:
I ran into this issue when building `fast-select` - I initially used other fast components in the template, which creates a weird circular dependency where we need fast-components to build fast-foundation, which is needed to build fast-components. This also means that the `fast-` tag "namespace" will need to be permanent, which could create conflicts for authors who want to compose their own components off of `fast-foundation`.

Reply:
Ahh yes, this will be resolved with a recent change similar to Tooltip where a prefix is given to createPopoverTemplate, but i failed to include that here in this example. Updated now.